commands/inventory/inventory.go

```diff
@@ -30,3 +33,19 @@ var (
 	inventoryAddC    = inventoryC.Command("add", "Add to inventory commands").Hide(hide)
 	inventoryRemoveC = inventoryC.Command("remove", "Remove from inventory commands").Hide(hide)
 )
+
+func formatTypeValue(acceptableTypeValues map[string][]string, input string) (*string, error) {
```

Comment:
Looks too complicated to be left without a comment.

Reply:
It looks like here you try to solve the search in the dictionary problem.

IMHO could be more natural to solve it with flat key - value:
```
var acceptableAgentTypes = map[string]string{
"pmm-agent": types.AgentTypePMMAgent,
strings.ToLower(types.AgentTypeName(types.AgentTypePMMAgent)): types.AgentTypePMMAgent,
"node-exporter": types.AgentTypeNodeExporter,
strings.ToLower(types.AgentTypeName(types.AgentTypeNodeExporter)): types.AgentTypeNodeExporter
...
}

func formatTypeValue(acceptableTypeValues map[string]string, input string) (*string, error) {
        if input == "" {
 		return nil, nil
 	}
        value, ok := acceptableTypeValues[strings.ToLower(input)]
        if !ok {
              return nil, fmt.Errorf("unexpected type value %q", input)
        }
        return &value, nil
}

This is faster in execution and easier to read if there is no other usage limitation.

Reply:
We need a better comment and maybe a better function name. Perhaps something that express we are getting the agent type name from a list of name **aliases**.

Reply:
@askomorokhov I wanted to make it readable here https://github.com/percona/pmm-admin/pull/94/files#diff-1c9adf29eb667fe733bf6d5dd8c0b512R38 to know what options are acceptable for some type in a row. Instead of looking for it in 30 lines we are able to see all options in 10 lines.

Reply:
@percona-csalguero it's not only for agent type, but we also use it for services and nodes too